src/lib/lib_weechat.py

```diff
@@ -51,9 +51,13 @@ def weechat_command(self, command):
             return 'Empty command'
         cmd = "*{}".format(command)
         hookenv.log("Calling fifo with: {}".format(cmd))
-        result = subprocess.check_output('echo {} > {}'.format(cmd, self.fifo_file),
-                                         shell=True)
-        return result.decode()
+        try:
+            with open(self.fifo_file,"w") as fifo:
+                print(cmd, file = fifo)
+        except:
+            return False
+        else:
```

Comment:
I don't think there's any reason for an 'else' here is there? Just try, catch an error and return false, and return true if you make it to the end of the function.